Content.Server/Traitor/Uplink/UplinkSystem.cs

```diff
@@ -1,82 +1,130 @@
-            store.AccountOwner = user;
-            store.Balance.Clear();
-            if (balance != null)
-            {
-                store.Balance.Clear();
-                _store.TryAddCurrency(new Dictionary<string, FixedPoint2> { { TelecrystalCurrencyPrototype, balance.Value } }, uplinkEntity.Value, store);
-            }
+        EnsureComp<UplinkComponent>(uplinkEntity.Value);
+
+        SetUplink(user, uplinkEntity.Value, balance);
 
-            // TODO add BUI. Currently can't be done outside of yaml -_-
+        // TODO add BUI. Currently can't be done outside of yaml -_-
+
+        return true;
+    }
 
-            return true;
+    /// <summary>
+    /// Configure TC for the uplink
+    /// </summary>
+    public void SetUplink(EntityUid user, EntityUid uplink, FixedPoint2? balance)
+    {
+        var store = EnsureComp<StoreComponent>(uplink);
+        store.AccountOwner = user;
+        store.Balance.Clear();
+        if (balance != null)
+        {
+            store.Balance.Clear();
+            _store.TryAddCurrency(new Dictionary<string, FixedPoint2> { { TelecrystalCurrencyPrototype, balance.Value } }, uplink, store);
         }
+    }
 
-        /// <summary>
-        /// Finds the entity that can hold an uplink for a user.
-        /// Usually this is a pda in their pda slot, but can also be in their hands. (but not pockets or inside bag, etc.)
-        /// </summary>
-        public EntityUid? FindUplinkTarget(EntityUid user)
+    /// <summary>
+    /// Implant an uplink as a fallback measure if the traitor had no PDA
+    /// </summary>
+    public bool ImplantUplink(EntityUid user, FixedPoint2? balance)
+    {
+        var implants = new List<string>(){FallbackUplinkImplant};
+
+        if (!_proto.TryIndex<ListingPrototype>(FallbackUplinkCatalog, out var catalog))
+            return false;
+
+        if (!catalog.Cost.TryGetValue(TelecrystalCurrencyPrototype, out var cost))
+            return false;
+
+        var implantCost = cost.Int();
+
+        _subdermalImplant.AddImplants(user, implants);
+
+        if (!_container.TryGetContainer(user, ImplanterComponent.ImplantSlotId, out var implantContainer))
+            return false;
+
+        foreach (var implant in implantContainer.ContainedEntities)
         {
-            // Try to find PDA in inventory
-            if (_inventorySystem.TryGetContainerSlotEnumerator(user, out var containerSlotEnumerator))
+            if (TryComp<StoreComponent>(implant, out var storeComp))
             {
-                while (containerSlotEnumerator.MoveNext(out var pdaUid))
-                {
-                    if (!pdaUid.ContainedEntity.HasValue) continue;
-
-                    if (HasComp<PdaComponent>(pdaUid.ContainedEntity.Value) || HasComp<StoreComponent>(pdaUid.ContainedEntity.Value))
-                        return pdaUid.ContainedEntity.Value;
-                }
+                SetUplink(user, implant, balance - implantCost);
```

Comment:
You might also want to make sure that (balance - implantCost) does not become negative if you have 0 starting balance for some reason.

Reply:
must kill renault to pay off your uplink loan :trollface:

Reply:
Should they even get an uplink at all, if their starting TC is less than the cost of the implant?

Reply:
Good question. It is probably fine if you still get one, even if you have no TC to use it with. That way you may still obtain some from other sources later.

Reply:
Balance now generously erases a potential debt if the traitor somehow started with 0 TC before force-buying the uplink, mostly to avoid potential weird-math-situations in other parts of the code